Source/SwiftLintBuiltInRules/Rules/Performance/EmptyCountRule.swift

```diff
@@ -31,31 +33,117 @@ struct EmptyCountRule: OptInRule {
     final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
-        private let operators: Set = ["==", "!=", ">", ">=", "<", "<="]
-
         override func visitPost(_ node: InfixOperatorExprSyntax) {
-            guard let operatorNode = node.operator.as(BinaryOperatorExprSyntax.self),
-                  let binaryOperator = operatorNode.operator.binaryOperator,
-                  operators.contains(binaryOperator) else {
+            guard node.hasBinaryOperator else {
                 return
             }
 
-            if let intExpr = node.rightOperand.as(IntegerLiteralExprSyntax.self), intExpr.isZero,
-               let position = node.leftOperand.countCallPosition(onlyAfterDot: configuration.onlyAfterDot) {
+            if let (_, position) = node.countNodeAndPosition(onlyAfterDot: configuration.onlyAfterDot) {
                 violations.append(position)
-                return
             }
+        }
+    }
 
-            if let intExpr = node.leftOperand.as(IntegerLiteralExprSyntax.self), intExpr.isZero,
-               let position = node.rightOperand.countCallPosition(onlyAfterDot: configuration.onlyAfterDot) {
-                violations.append(position)
-                return
+    final class Rewriter: ViolationsSyntaxRewriter {
+        private let configuration: EmptyCountConfiguration
+
+        init(configuration: EmptyCountConfiguration,
+             locationConverter: SourceLocationConverter,
+             disabledRegions: [SourceRange]) {
+            self.configuration = configuration
+            super.init(locationConverter: locationConverter, disabledRegions: disabledRegions)
+        }
+
+        override func visit(_ node: InfixOperatorExprSyntax) -> ExprSyntax {
+            guard node.hasBinaryOperator else {
+                return super.visit(node)
+            }
+
+            if let (count, position) = node.countNodeAndPosition(onlyAfterDot: configuration.onlyAfterDot) {
+                let newNode: ExprSyntax? =
+                    if let count = count.as(MemberAccessExprSyntax.self) {
+                        count.with(\.declName.baseName, "isEmpty").trimmed.as(ExprSyntax.self)
+                    } else if let count = count.as(DeclReferenceExprSyntax.self) {
+                        count.with(\.baseName, "isEmpty").trimmed.as(ExprSyntax.self)
+                    } else {
+                        nil
+                    }
+
+                if let newNode, let binaryOperator = node.binaryOperator {
+                    correctionPositions.append(position)
+                    return
+                        if ["!=", "<", ">"].contains(binaryOperator) {
+                            newNode.negated
+                                .withTrivia(from: node)
+                        } else {
+                            ExprSyntax(newNode)
+                                .withTrivia(from: node)
+                        }
+                } else {
+                    let left =
```

Comment:
There still seems to be no examples which check this `else` branch. Could you please add some?

Reply:
This `else` branch was removed and one example was added to test this case. Please see https://github.com/realm/SwiftLint/pull/5409#discussion_r1468677885.